src/workflows/pyfr/common/steps/Visualization/Start/index.js

```diff
@@ -55,11 +55,22 @@ const VisualizationStart = React.createClass({
     const dataDir = this.props.simulation.steps.Visualization.metadata.dataDir;
     const fileName = this.props.simulation.steps.Visualization.metadata.fileName;
     const payload = {
-      cluster: { _id: this.state[this.state.serverType].profile },
       dataDir, // where the output for the sim will be
       fileName, // the file to load
       sessionKey, // for pvw, we use this later for connecting
     };
+
+    if (this.state.serverType === 'Traditional') {
+      payload.cluster = { _id: this.state[this.state.serverType].profile };
+    } else if (this.state.serverType === 'EC2') {
+      payload.cluster = {
+        type: 'ec2',
+        machineType: this.state[this.state.serverType].machine,
+        clusterSize: this.state[this.state.serverType].clusterSize,
+        profileId: this.state[this.state.serverType]._id,
+      };
```

Comment:
Is there away we could avoid this duplicate code block, it appears in the PyFR workflow as well? We are also missing cluster name from the payload, which is a good reason to try a avoid the duplication is we can.